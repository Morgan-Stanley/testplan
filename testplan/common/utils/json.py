import json

_USE_RAPIDJSON = False

try:
    import rapidjson
except ImportError:
    pass
else:
    _USE_RAPIDJSON = True


def json_loads(data: str):
    if _USE_RAPIDJSON:
        # being explicit here
        return rapidjson.loads(data, number_mode=rapidjson.NM_NAN)
    else:
        return json.loads(data)


def json_dumps(data, indent_2=False, default=None) -> str:
    if _USE_RAPIDJSON:
        # being explicit here
        return rapidjson.dumps(
            data,
            indent=2 if indent_2 else None,
            default=default,
            number_mode=rapidjson.NM_NAN,
        )
    else:
        if default:

            class _E(json.JSONEncoder):
                def default(self, o):
                    return default(o)

        else:
            _E = None
        return json.dumps(data, cls=_E, indent=2 if indent_2 else None)
