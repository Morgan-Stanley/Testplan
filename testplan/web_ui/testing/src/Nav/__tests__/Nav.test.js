import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import Nav from '../Nav';

const TESTPLAN_REPORT =
  require('../../__tests__/mocks/documents/TESTPLAN_REPORT_1.json');

const defaultProps = {
  report: TESTPLAN_REPORT,
  selected: [TESTPLAN_REPORT],
};

describe('Nav', () => {

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterEach(() => {
    // Resume style injection once test is finished.
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('shallow renders and matches snapshot', () => {
    const rendered = shallow(<Nav {...defaultProps} />);
    expect(rendered).toMatchSnapshot();
  });

});

