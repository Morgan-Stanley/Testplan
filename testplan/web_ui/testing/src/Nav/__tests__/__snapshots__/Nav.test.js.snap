// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav shallow renders and matches snapshot 1`] = `
<Fragment>
  <NavBreadcrumbs
    entries={
      Array [
        Object {
          "name": "Sample Testplan",
          "status": "failed",
          "uid": "520a92e4-325e-4077-93e6-55d7091a3f83",
        },
      ]
    }
  />
  <NavList
    breadcrumbLength={1}
    entries={
      Array [
        Object {
          "category": "multitest",
          "name": "Primary",
          "status": "failed",
          "tags": Object {
            "simple": Array [
              "server",
            ],
          },
          "type": "TestGroupReport",
          "uid": "21739167-b30f-4c13-a315-ef6ae52fd1f7",
        },
        Object {
          "category": "multitest",
          "name": "Secondary",
          "status": "passed",
          "tags": Object {},
          "type": "TestGroupReport",
          "uid": "8c3c7e6b-48e8-40cd-86db-8c8aed2592c8",
        },
      ]
    }
  />
</Fragment>
`;
